checks.d/powerdns_authoritative.py

```diff
@@ -0,0 +1,69 @@
+# stdlib
+from collections import namedtuple
+
+# Datadog
+from checks import AgentCheck
+
+# 3p
+import requests
+
+class PowerDNSAuthoritativeCheck(AgentCheck):
+    # See https://doc.powerdns.com/md/authoritative/performance/#performance-monitoring for metrics explanation
+
+    SERVICE_CHECK_NAME = 'powerdns.authoritative.can_connect'
+
+    def check(self, instance):
+        config, tags = self._get_config(instance)
+        stats = self._get_pdns_stats(config)
+        for stat in stats:
+            self.log.debug('powerdns.authoritative.{}:{}'.format(stat['name'], stat['value']))
+
+            if 'status' in stat['name']:
+                self.gauge('powerdns.authoritative.{}'.format(stat['name']), float(stat['value']), tags=tags)
+            elif 'size' in stat['name']:
+                self.gauge('powerdns.authoritative.{}'.format(stat['name']), float(stat['value']), tags=tags)
+            elif 'usage' in stat['name']:
+                self.gauge('powerdns.authoritative.{}'.format(stat['name']), float(stat['value']), tags=tags)
+            elif 'msec' in stat['name']:
+                self.gauge('powerdns.authoritative.{}'.format(stat['name']), float(stat['value']), tags=tags)
+            elif 'time' in stat['name']:
+                self.gauge('powerdns.authoritative.{}'.format(stat['name']), float(stat['value']), tags=tags)
+            elif 'latency' in stat['name']:
+                self.gauge('powerdns.authoritative.{}'.format(stat['name']), float(stat['value']), tags=tags)
+            else:
+                self.rate('powerdns.authoritative.{}'.format(stat['name']), float(stat['value']), tags=tags)
```

Comment:
Can you simplify the logic in this conditional? We usually like to list the metrics we gather within the check within a constant at the top of the class, like `RATE_METRICS` and `GAUGE_METRICS`. 

This adds to the clarity of the check, because you can tell what is being gathered, but it also ensures that a single check isn't sending a potentially unlimited number of different metrics.